cmd/admin-bucket-handlers.go

```diff
@@ -169,6 +170,9 @@ func (a adminAPIHandlers) SetRemoteTargetHandler(w http.ResponseWriter, r *http.
 		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErrWithErr(ErrAdminConfigBadJSON, err), r.URL)
 		return
 	}
+	if strings.Contains(target.Endpoint, "storage.googleapis.com") {
+		target.Endpoint = host
+	}
```

Comment:
it is better to generalize it @poornas i.e `net.SplitHostPort(target.Endpoint)`

Reply:
ok, will do

Reply:
net.SplitHostPort is in /usr/local/go/src/net - it has to be done outside

Reply:
the fix is not going to work since net.SplitHostPort would reject and your code won't even reach here. @poornas  as the target.Endpoint doesn't contain port by default. 

Better way is related to the port requirement basically if the port is missing assume port 443 or port 80 based on the target.Secure parameter

Reply:
fixed